internal/services/notification/runstatus.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2023 Sorint.lab
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package notification
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"agola.io/agola/internal/errors"
+	"agola.io/agola/internal/services/common"
+	rstypes "agola.io/agola/services/runservice/types"
+)
+
+var jsonContent = "application/json"
+
+const webhookVersion = 1
+
+type Webhook struct {
+	// version of the webhook struct data
+	Version uint64 `json:"version"`
+
+	// path of the project
+	ProjectPath string `json:"project_path"`
+
+	// current run status
+	Run *Run `json:"run"`
+
+	Secret string `json:"secret"`
```

Comment:
So we are just returning the provided secret. This is like gitlab does. But it's less secure.

Instead gitea and github provides a signature of the payload in an header and the client can validate this signature agains the secret. I'll prefer this way.